### PR TITLE
New RFC: External Control Plane for Gluster

### PR DESCRIPTION
**Pull request by aravindavk**
Signed-off-by: Aravinda Vishwanathapura <aravinda@kadalu.io>

### CODE REVIEW

File: text/0000-external-control-plane-gluster.md [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+devices.
+
+Log in to the Control server by running the following command, and the
+token will be saved in `~/.gluster/token.json` on a successful login.
+
+All consecutive commands will use the token saved in the file
+mentioned above. To logout, run the following command.
+
+```
+$ glustercli login
```
Comment (amarts) [inline text/0000-external-control-plane-gluster.md]: you want to call this tool `glustercli` or something else? like `moana` (means sea/kadalu/ocean) or something similar?

**Review by amarts — COMMENTED** [review]
A great proposal! No issues seen in first round review. Will check again in few days!

**Comment by sankarshanmukhopadhyay** [comment]
LGTM

In a future iteration we should solicit inputs on how authorization and access of users can be integrated with existing IAM pieces. Traditionally, Gluster has not documented any specific ways instead, leaving that to the directory services or, IAM frameworks to decide.

**Review by amarts — APPROVED** [review]

**Comment by howels** [comment]
Suggestion for increased adoption - would it be possible to migrate KIS PVs from being Heketi-managed to being Kadalu-managed? If a migration route was possible this would make it feasible for users with running workloads to contemplate a switch. Not always possible to remove and re-add the storage.

**Comment by aravindavk** [comment]
> Suggestion for increased adoption - would it be possible to migrate KIS PVs from being Heketi-managed to being Kadalu-managed? If a migration route was possible this would make it feasible for users with running workloads to contemplate a switch. Not always possible to remove and re-add the storage.

Existing Gluster Cluster can be migrated to use this External Control plane without changing the Filesystem. This solution is for non-Kubernetes use cases. For the Kubernetes use case, Kadalu already supports using external Gluster Volumes(Heketi or independently managed Gluster Cluster). See here: https://kadalu.io/docs/k8s-storage/latest/external-gluster-storage

**Comment by howels** [comment]
> > Suggestion for increased adoption - would it be possible to migrate KIS PVs from being Heketi-managed to being Kadalu-managed? If a migration route was possible this would make it feasible for users with running workloads to contemplate a switch. Not always possible to remove and re-add the storage.
> 
> Existing Gluster Cluster can be migrated to use this External Control plane without changing the Filesystem. This solution is for non-Kubernetes use cases. For the Kubernetes use case, Kadalu already supports using external Gluster Volumes(Heketi or independently managed Gluster Cluster). See here: https://kadalu.io/docs/k8s-storage/latest/external-gluster-storage

Thanks. We have a mix of Kubernetes and non-Kubernetes volumes in our GlusterFS clusters.

Out of interest why do you recommend sharing Gluster volumes for multiple PV? We have some errors with metadata filling up in our Gluster-backed PV and blocking writes at present and would be anxious about a similar issue stopping multiple PVs at once.

**Pull request merged**